assets/javascripts/discourse/components/assigned-topic-list-item.js.es6

```diff
@@ -0,0 +1,187 @@
+import DiscourseURL from "discourse/lib/url";
+import computed from "ember-addons/ember-computed-decorators";
+import { wantsNewWindow } from "discourse/lib/intercept-click";
+
+export function showEntrance(e) {
+  let target = $(e.target);
+
+  if (target.hasClass("posts-map") || target.parents(".posts-map").length > 0) {
+    if (target.prop("tagName") !== "A") {
+      target = target.find("a");
+      if (target.length === 0) {
+        target = target.end();
+      }
+    }
+
+    this.appEvents.trigger("topic-entrance:show", {
+      topic: this.get("topic"),
+      position: target.offset()
+    });
+    return false;
+  }
+}
+
+export function navigateToTopic(topic, href) {
+  this.appEvents.trigger("header:update-topic", topic);
+  DiscourseURL.routeTo(href || topic.get("url"));
+  return false;
+}
+
+export default Ember.Component.extend({
```

Comment:
Can we not just inherit off core here and reuse the majority of the implementation?